src/structures/idt.rs

```diff
@@ -918,6 +931,60 @@ bitflags! {
     }
 }
 
+// https://wiki.osdev.org/Exceptions#Selector_Error_Code
+/// Describes a segment selector error code.
+#[derive(Clone, Copy)]
+#[repr(transparent)]
+pub struct SelectorErrorCode {
+    flags: u64,
+}
+
+impl SelectorErrorCode {
+    ///  If this flag is set, it indicates that the exception originated externally to the processor
+    pub fn external(&self) -> bool {
+        self.flags.get_bit(0)
+    }
+
+    /// The descriptor table where the exception occurred.
+    pub fn descriptor_table(&self) -> DescriptorTable {
+        match self.flags.get_bits(1..3) {
+            0b00 => DescriptorTable::Gdt,
+            0b01 => DescriptorTable::Idt,
+            0b10 => DescriptorTable::Ldt,
+            0b11 => DescriptorTable::Idt,
+            _ => unreachable!(),
+        }
+    }
+
+    /// The index of the descriptor table.
+    pub fn index(&self) -> u64 {
+        self.flags.get_bits(3..16)
+    }
+}
```

Comment:
Would it make sense to add an `is_null()` method to `SelectorErrorCode`? The `#SS` and `#GP` sometimes return an actual selector error code, and sometimes return 0 depending on the exact reason for the error.